gengokit/httptransport/templates.go

```diff
@@ -84,13 +84,14 @@ var ClientEncodeTemplate = `
 
 		// Set the body parameters
 		var buf bytes.Buffer
-		toRet := map[string]interface{}{
+		toRet := request.(*pb.{{GoName $binding.Parent.RequestType}})
 		{{- range $field := $binding.Fields -}}
 			{{if eq $field.Location "body"}}
-				"{{$field.CamelName}}" : req.{{$field.CamelName}},
+				{{/* Only set the fields which should be the body, so all
```

Comment:
What's going on is that we are making a request to the server, and that request contains fields. However, an endpoint may only want *certain* fields in the body of the POST to that server.

At this point in code generation, we have the struct form of the data which should be sent to the server. What I'm trying to convey is that by creating a new struct and only copying the fields which should be sent as part of the body into it, we can marshal the new struct to string form and get a json message which only contains the fields we copied, which are the fields we want to send in the body.

The potential problem with this approach is that the unset fields still contain their default value, which may be encoded and sent in the body. However, since all generated struct fields have struct tags indicating `omitempty`, struct fields with uninitialized values won't show up in the json. Thus, the one potential problem with the approch is avoided.

---------

That's pretty long though, so I used the short comment that you see there, as terse as it is. Do you have any recommendations for a better comment?